src/controllers/authController.js

```diff
@@ -0,0 +1,43 @@
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const User = require('../models/User.model.js');
+require('env2')('config.env');
+
+const index = (req, res) => {
+  res.render('auth');
+};
+
+
+const register = (req, res) => {
```

Comment:
This looks good to me but it has become really long. It would be great if you wrote the separate functions and referenced them in each promise i.e. 
```
.then(functionName)
.then(functionName)
```